bin/prepare_feedback.py

```diff
@@ -0,0 +1,173 @@
+    # rename topic for all tutorials in a topic
+    for topic in new_topics:
+        df.topic = (df
+            .topic
+            .replace(to_replace=topic, value=new_topics[topic]))
+    # rename some tutorials
+    for tuto in new_tutorial_name:
+        df.loc[df.tutorial == tuto, 'tutorial'] = new_tutorial_name[tuto]
+    return df
+
+
+def extract_topic_tutorial_name(df):
+    '''Extract topic from tutorial name
+
+    :param df: dataframe
+    '''
+    df = df.copy()
+    new = df['tutorial_topic'].str[::-1].str.split('(', n = 1, expand = True)
+    df["tutorial"]= new[1].str[::-1].str[:-1]
+    df["topic"]= new[0].str[::-1].str[:-1]
+    return df
+
+
+def prepare_feedback(url, out_file):
+    '''Get and prepare feedback CSV file
+
+    :param url: URL to Google sheet with feedback answers
+    :param out_file: Path to output file
+    '''
+    df = (pd.read_csv(url, sep='\t')
+        # rename column
+        .rename(columns = {'Timestamp': 'timestamp',
+                            'How much did you like this tutorial?': 'note',
+                            'What did you like?': 'pro',
+                            'What could be improved?': 'con',
+                            'Tutorial': 'tutorial_topic',
+                            'Your feedback is always anonymous. Also make it confidential (only visible to admins)?': 'anonymous'})
+        # extract topic from tutorial name
+        .pipe(extract_topic_tutorial_name)
+        # remove rows with NaN on note, pro and con
+        .dropna(subset=['note', 'pro', 'con'], how='all')
+        # replace NaN in note by 0
+        .fillna(value={'note': 0})
+        # fill other NaN by empty string
+        .fillna('')
+        # format 
+        .assign(
+            #note to integer
+            note=lambda x: x['note'].astype(int),
+            # format pro and con to string
+            pro=lambda x: x['pro'].astype(str),
+            con=lambda x: x['con'].astype(str),
+            # extract month and date
+            timestamp=lambda x: pd.to_datetime(x['timestamp'], dayfirst=True))
+        .assign(
+            month=lambda x: x['timestamp'].dt.strftime("%Y-%m"),
+            date=lambda x: x['timestamp'].dt.strftime("%Y-%m-%d"))
+        # change topic for some tutorials
+        .pipe(fix_tutorial_info)
+        # remove extra columns
+        .drop(columns=["tutorial_topic", "timestamp"])
+        # remove some rows 
+        .query('topic != "TESTIN"')
+        .query('topic != "tes"')
+        .query('topic != ""')
+        # remove tutorials
+        .query('tutorial != "RNA-seq counts to genes and pathways"')
+        .query('tutorial != "Galaxy Tool Management"')
+        .query('tutorial != "Recording Job Metrics"')
+        #
+        .query('pro != "TESTING"')
+        .to_csv(out_file))
+
+
+if __name__ == '__main__':
+    url = 'https://docs.google.com/spreadsheets/d/1NfZhi5Jav7kl9zFCkeb7rIC2F8xW1isruv1TeO4WpNI/export?format=tsv'
```

Comment:
this would require making it public, right? that url 404s for me.

If we want to do that, should we remove the people who didn't consent to their feedback being public? Should we remove the checkbox completely and just write in bold text "This will be public"?

Reply:
@shiltemann reminds me that we've already done the "it's always public", shall we just remove those rows from the table and make it public?

(If we're making it public I guess we could just embed the google graphs directly and wouldn't need the round trip?)

Reply:
Interesting, It works for me within a private window.
Not sure what to do with the private feedback (no private information are stored), as they are quite a lot. Currently I just used the notes from the private feedback, not the free text...
I am not sure to understand what do you mean by the google graphs?

Reply:
> Interesting, It works for me within a private window.

seems temporary, it's working now :)

>  Not sure what to do with the private feedback (no private information are stored)

it feels quite wrong to expose it just since they didn't consent (yeah they probably misunderstood/misread the question, but, still, we hadn't been exposing this previously)

>  I am not sure to understand what do you mean by the google graphs?

if we're going to publicly expose that feedback table, then we could make many of the same graphs in google sheets probably, right? Score by topic (since that's in the tutorial name), score by tutorial.

And then we could just embed those directly? It would save us the bother of setting up a cron job, we could remove the python, the jekyll template would just be a couple iframes

![image](https://user-images.githubusercontent.com/458683/126144641-69b3e5a9-d21a-4a01-b9d1-4decdfc12f7f.png)

Reply:
we'd just have to handle the renaming/reshaping directly in the tables.

I think we can also publish graphs without publishing the entire spreadsheet which could help address the 'private' data? (But we wouldn't get the nice pro/con lists you've gone to the trouble to extract)

Reply:
I tried before (at the beginning) to do it within Google sheet but it did not render nicely (that is why I spent quite some times polishing data and creating these graphs first within Jupyter notebooks and then here). But feel free to give it a try

Reply:
I had not remembered that! No, not better, just thought maybe it would be easier. I trust your experience doing that, that that path isn't worth going down!